expression/scalar_function.go

```diff
@@ -245,8 +245,14 @@ func (sf *ScalarFunction) HashCode(sc *stmtctx.StatementContext) []byte {
 }
 
 // ResolveIndices implements Expression interface.
-func (sf *ScalarFunction) ResolveIndices(schema *Schema) {
+func (sf *ScalarFunction) ResolveIndices(schema *Schema) Expression {
+	newSf := sf.Clone()
```

Comment:
we can:
```go
newSf := sf.Clone()
// resolve index on "newSf"
return newSf
```

Reply:
It's just what i've written here?

Reply:
But, its args are expression, may not be column.
If we call args' `ResolveIndices`, it will generate new copies of args.
If we call args' `resolveIndices`, then there still needs a method on interface named `resolveIndices`.

Reply:
Just found that there is not a function to set an argument for an `Expression`.... This way is ok for now.

Reply:
Yes, i think it should be like this. The scalar function should not change its args. Since expression have `HashCode`. Set arg may break the hash code.

Reply:
Yes, but we still can implement a function named `SetArg` to change its arguments, and reset its hash code if necessary in that function. That's part of the implementation detail about how to implement a `SetArg` function. We can discuss it later when we really need a `SetArg` function.